src/fu-engine.c

```diff
@@ -1149,37 +1169,50 @@ fu_engine_check_requirement_firmware (FuEngine *self, XbNode *req,
 
 	/* child version */
 	if (g_strcmp0 (xb_node_get_text (req), "not-child") == 0)
-		return fu_engine_check_requirement_not_child (self, req, device, error);
+		return fu_engine_check_requirement_not_child (self, req, device_actual, error);
 
 	/* another device */
 	if (fwupd_guid_is_valid (xb_node_get_text (req))) {
 		const gchar *guid = xb_node_get_text (req);
 		const gchar *version;
-		g_autoptr(FuDevice) device2 = NULL;
 
 		/* find if the other device exists */
-		device2 = fu_device_list_get_by_guid (self->device_list, guid, error);
-		if (device2 == NULL)
-			return FALSE;
+		if (depth == G_MAXUINT64) {
+			device_actual = fu_device_list_get_by_guid (self->device_list, guid, error);
```

Comment:
doesn't this need to be a g_set_object call to since you took a reference above?

Reply:
Good catch! Fixed.